frontend/public/components/marketplace/kubernetes-marketplace-items.jsx

```diff
@@ -0,0 +1,298 @@
+  constructor(props) {
+    super(props);
+    const {items} = this.props;
+
+    // Filters are populated based on white list of properties
+    const filters = getFilters(items, filterCategories);
+
+    this.state = {
+      filters,
+      ...this.getFilterItemsAndCounts(filters),
+    };
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    const {filters} = this.state;
+    const {items} = this.props;
+    if ((items === prevProps.items) && (filters === prevState.filters)) {
+      return;
+    }
+
+    const updatedState = {};
+
+    // Update filters to match new items
+    if (items !== prevProps.items) {
+      const newFilters = getFilters(items, filterCategories);
+      // Apply current filter state to new filters
+      updatedState.filters = copyActiveFilters(newFilters, filters);
+    }
+
+    // If only filters have changed, update filter items and filter counts
+    if (updatedState.filters || filters !== prevState.filters) {
+      Object.assign(updatedState, this.getFilterItemsAndCounts(updatedState.filters || filters));
+    }
+
+    this.setState(updatedState);
+  }
+
+  getFilterItemsAndCounts(filters) {
+    const itemsAndCounts = {
+      filteredItems: {},
+      filterCounts: {},
+    };
+
+    // Filter items by each filter category individually
+    // Example:
+    //  filteredByCategory = {
+    //    provider: [/*array of items filtered by provider*/],
+    //    healthIndex: [/*array of items filtered by healthIndex*/],
+    //  };
+    const filteredByCategory = this.filterByCategory(filters);
+
+    // The set intersection of the above arrays
+    // results in an array of items filtered by all categories
+    const filteredItems = _.intersection(..._.toArray(filteredByCategory));
+    itemsAndCounts.filteredItems = _.sortBy(this.filterByName(filteredItems, filters), 'name');
+
+    // Get counts for filtered items in each category
+    itemsAndCounts.filterCounts = this.getFilterCountsFromCategories(filters, filteredByCategory);
+
+    return itemsAndCounts;
+  }
+
+  filterByCategory(filters) {
+    const {items} = this.props;
+
+    // Filters as follows:
+    //   Active filters in the same category result in the union of items shown
+    //   Active filters in separate categories result in the intersection
+    const filteredByCategory = _.reduce(filters, (filtered, category, key) => {
+      if (key === 'name') {
+        return filtered;
+      }
+
+      // Apply each filter individually to items
+      let categoryActive = false;
+      const categoryItems = _.reduce(category, (accum, elem) => {
+        if (elem.active) {
+          categoryActive = true;
+          accum.push(_.filter(items, [ key, elem.value ]));
+        }
+        return accum;
+      }, []);
+
+      // Filter items only if at least one filter in the category is active
+      // Run union on items only if filter category is active
+      filtered[key] = categoryActive ? _.union(...categoryItems) : items;
```

Comment:
Can you check if this does the same thing? I think it's a little simpler.

```js
    const filteredByCategory = _.reduce(filters, (filtered, category, key) => {
      if (key === 'name') {
        return filtered;
      }
      
      const activeFilters = _.filter(category, 'active');
      if (activeFilters.length) {
        const values = _.map(activeFilters, 'value');
        filtered[key] _.filter(items, item => values.includes(item[key]));
      } else {
        filtered[key] = items;
      }

      return filtered;
    });
```

Reply:
I agree, looks simpler to me too! Works with some finesse:
```
_.reduce(filters, (filtered, category, key) => {
  if (key === 'keyword') {
    return filtered;
  }

  const activeFilters = _.filter(category, 'active');
  if (activeFilters.length) {
    const values = _.map(activeFilters, 'value');
    filtered[key] = _.filter(items, item => values.includes(item[key]));
  } else {
    filtered[key] = items;
  }

  return filtered;
}, {});
```